0029-stream/0029-stream.md

```diff
@@ -170,19 +172,28 @@ Incoming packets can either be associated with an existing connection, or, for s
 
-When a client connects to a server, they MUST communicate their ILP address to the server using a `ConnectionNewAddress` frame.
+When a client connects to a server, the client MAY communicate its ILP Address to the server using a  `ConnectionNewAddress` frame. This allows the client to function as a receiver on the Connection. Without this frame, a server would not have a destination address to send packets to a client.
 
-Either endpoint MAY change their ILP address at any point during a connection by sending a `ConnectionNewAddress` frame. To ensure the new address is received and acknowledged, implementations MAY choose to send these frames only in ILP Prepare packets.
+In the most advanced scenarios where clients and servers are both capable of simultaneously sending and receiving, the `ConnectionNewAddress` frame enables bi-directional payment flows. However, the frame is OPTIONAL because clients are commonly not capable of receiving. For example, a client may not be accessible from a routable ILP Address.
 
-Implementations SHOULD wait for a valid response (encrypted with the same shared secret) from the new address to validate the new path. STREAM uses the authenticated request/response packets in lieu of [QUIC's explicit Path Validation](https://quicwg.github.io/base-drafts/draft-ietf-quic-transport.html#rfc.section.6.7). Implementations SHOULD refrain from sending large numbers of packets or large amounts of data to a new ILP address before validating the path to avoid being tricked into participating in a Denial of Service (DoS) attack on a third-party endpoint.
+#### 4.3.2. Connection Migration (Receivers Only)
 
-#### 4.3.1. Connection Asset Details
+If an endpoint supports receiving, then the endpoint MAY change its ILP Address at any point during a connection by sending a `ConnectionNewAddress` frame. To ensure the new address is received and acknowledged, implementations MAY choose to send these frames only in ILP Prepare packets, although certain connections may not support this (e.g., a receiver emitting this frame to a non-receiving sender will only be able to propagate this frame in a fulfill or reject packet).
 
-Each endpoint MAY expose their asset details by sending a `ConnectionAssetDetails` frame. This frame is optional because some use-cases do not require it.
+Senders encountering this frame SHOULD wait for a separate, valid request/response (encrypted with the same shared secret) from the new address to validate the new path. STREAM relies upon this authenticated request/response packet flow in lieu of [QUIC's explicit Path Validation](https://quicwg.github.io/base-drafts/draft-ietf-quic-transport.html#rfc.section.6.7), so implementations SHOULD refrain from sending large numbers of packets or large amounts of data to a new ILP address before validating the path. For example, this might help avoid being tricked into participating in a Denial of Service (DoS) attack on a third-party endpoint.
```

Comment:
is there some kind of packet that is recommended to probe a new connection address?

Reply:
I'm not sure which frame would be best. Though it's occurring to me that it may not matter -- even sending 0 frames would probably be fine. As long as you get a validly encrypted StreamPacket back from the other side, I think this satisfies the validation check. In other words, the sender just needs to validate that the new receiver address can decrypt the packets. If it can, it's reasonable for the client to assume the receiver wants those packets.

Not totally sure that's valid thinking though - did you have a different understanding here?